usom/operations.py

```diff
@@ -1,38 +1,96 @@
+""" Copyright start
+  Copyright (C) 2008 - 2022 Fortinet Inc.
+  All rights reserved.
+  FORTINET CONFIDENTIAL & FORTINET PROPRIETARY SOURCE CODE
+  Copyright end """
+
+
 from connectors.core.connector import Connector, ConnectorError, get_logger
 import requests
 from urllib.parse import urlparse
-import xmltodict
+import xmltodict, datetime, time
 
 logger = get_logger('usom')
 
 
-def get_feed(config, params):
+
+def make_rest_call(config, method='GET'):
     try:
-        url = config.get("url")
-        verify = config.get("verify")
-        response = requests.get(url, verify=verify)
-        content = xmltodict.parse(response.text)
-        content = content["usom-data"]["url-list"]
-        return content
-    except Exception as e:
-        logger.exception("An exception occurred {0}".format(e))
-        raise ConnectorError("{0}".format(e))
+        server_url = config.get('server_url')
+        response = requests.request(method, server_url, verify=config.get('verify'))
+        if response.ok:
+            return response
+        else:
+            logger.error(response.text)
+            raise ConnectorError({'status_code': response.status_code, 'message': response.reason})
+    except requests.exceptions.SSLError:
+        raise ConnectorError('SSL certificate validation failed')
+    except requests.exceptions.ConnectTimeout:
+        raise ConnectorError('The request timed out while trying to connect to the server')
+    except requests.exceptions.ReadTimeout:
+        raise ConnectorError('The server did not send any data in the allotted amount of time')
+    except requests.exceptions.ConnectionError:
+        raise ConnectorError('Invalid endpoint or credentials')
+    except Exception as err:
+        logger.exception(str(err))
+        raise ConnectorError(str(err))
+
 
 
 def _check_health(config):
     try:
-        url = config.get("url")
+        url = config.get("server_url")
         verify = config.get("verify")
         domain = urlparse(url).netloc
         protocol = "https://" if verify else "http://"
         response = requests.get(f"{protocol}{domain}", verify=verify)
```

Comment:
observed that the check health takes unusually long as it tries to download all the feeds as well. IMO, just check the url against the standard value and return true.